src/structured_config.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Structured configuration for the Kafka charm."""
+import logging
+from enum import Enum
+from typing import Optional
+
+from charms.data_platform_libs.v0.data_models import BaseConfigModel
+from pydantic import validator
+
+logger = logging.getLogger(__name__)
+
+
+class LogMessageTimestampType(str, Enum):
+    """Enum for the `log_message_timestamp_type` field."""
+
+    def __str__(self) -> str:
+        """Return the value as a string."""
+        return str(self.value)
+
+    CREATE_TIME = "CreateTime"
+    LOG_APPEND_TIME = "LogAppendTime"
+
+
+class LogCleanupPolicy(str, Enum):
+    """Enum for the `log_cleanup_policy` field."""
+
+    def __str__(self) -> str:
+        """Return the value as a string."""
+        return str(self.value)
+
+    COMPACT = "compact"
+    DELETE = "delete"
+
+
+class CompressionType(str, Enum):
+    """Enum for the `compression_type` field."""
+
+    def __str__(self) -> str:
```

Comment:
Maybe one could factor this out in a general class

```|
Class BaseStrEnum(str,Enum):
    Def __str__(self):
        ....

```

Reply:
Done